packages/workflow/src/Interfaces.ts

```diff
@@ -1366,6 +1382,17 @@ export interface ResourceMapperTypeOptions {
 	};
 }
 
+// Enforce at least one of resourceMapperMethod or localResourceMapperMethod
+export type ResourceMapperTypeOptions =
+	| (ResourceMapperTypeOptionsBase & {
+			resourceMapperMethod: string;
+			localResourceMapperMethod?: never;
+	  })
+	| (ResourceMapperTypeOptionsBase & {
+			localResourceMapperMethod: string;
+			resourceMapperMethod?: never;
+	  });
```

Comment:
How about this:

```suggestion
export type ResourceMapperTypeOptionsLocal = {
	resourceMapperMethod: string;
	localResourceMapperMethod?: never; // Explicitly disallows this property
};

export type ResourceMapperTypeOptionsExternal = {
	localResourceMapperMethod: string;
	resourceMapperMethod?: never; // Explicitly disallows this property
};

export type ResourceMapperTypeOptions = ResourceMapperTypeOptionsBase &
	(ResourceMapperTypeOptionsLocal | ResourceMapperTypeOptionsExternal);

```

It took me some time to parse the `never` here so I added the comment. Even made a [playground](https://www.typescriptlang.org/play/?#code/C4TwDgpgBAglC8UDeUAeAuKBGKBfAUPqJFAEILJQiYBMANFAF4D8mAdhAG4QBOehxaAGEKKRpgDMDEKygduPAvgiowAex7AogqACUIAZzUBXHgGMIAWQCGYSDwAq4CAHkwwAJZq2BinABkUAAU5AA+UEIAlADchGbeBloAtpj6RqYWNna8TpBungmiaJhY0rQE8T5abKmGJuZWtva5ru5ePkUY2GVQEhUJWmq16Q1Zzc757b6IKF2lTOX4lYlQYMP1mU05E22FM8XdC739VVAAjusZjdmOOwUd+3M99Ed9SwNQPJejW7d5uw9KE8qJIGOJjvggA) to verify my understanding, it's a nice solution :)